### PR TITLE
chore: isolate recharts to an untrusted iframe

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/22139

Sec review: https://github.com/brave/reviews/issues/1442

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

The portfolio graph chart and the asset price charts should continue to function without regressions.

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/portfolio-overview-chart/portfolio-overview-chart.tsx [hunk 1]
```diff
@@ -25,21 +31,51 @@ export const PortfolioOverviewChart: React.FC<Props> = ({
...
-        isLoading={hasZeroBalance ? false : isLoading}
-        isDisabled={hasZeroBalance}
+      <iframe
+        width={'100%'}
+        height={'130px'}
+        frameBorder={0}
+        src={`chrome-untrusted://line-chart-display${
+          isLoading ? '' : `?${encodedPriceData}`
+        }`}
+        sandbox='allow-scripts allow-same-origin'
```
Comment (diracdeltas) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/portfolio-overview-chart/portfolio-overview-chart.tsx:78]: not really an issue but just curious why allow-same-origin is needed?

Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/portfolio-overview-chart/portfolio-overview-chart.tsx]: Not needed. Removed. Thanks for catching this

File: components/brave_wallet_ui/line_chart/components/custom_tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  if (active && payload && payload[0].value) {
+    // Computed
+    const xLeftCoordinate = Math.trunc(coordinate?.x ?? 0)
+    const viewBoxWidth = Math.trunc(viewBox?.width ?? 0)
+    const xRightCoordinate = xLeftCoordinate - viewBoxWidth
+    const isEndOrMiddle = xRightCoordinate >= -62 ? 'end' : 'middle'
+    const labelPosition = xLeftCoordinate <= 62 ? 'start' : isEndOrMiddle
+    const middleEndTranslate =
+      xRightCoordinate >= 8 ? 0 : Math.abs(xRightCoordinate) - 4
```
Comment (kdenhartog) [inline components/brave_wallet_ui/line_chart/components/custom_tooltip.tsx:51]: Nit: some of these numbers seem a bit confusing as to why they're here. A more detailed comment could help here.

File: components/brave_wallet_ui/line_chart/line_chart.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<!--Copyright (c) 2023 The Brave Authors. All rights reserved.-->
+<!--This Source Code Form is subject to the terms of the Mozilla Public-->
+<!--License, v. 2.0. If a copy of the MPL was not distributed with this file,-->
+<!--you can obtain one at http://mozilla.org/MPL/2.0/.-->
+
+<!doctype html>
+<head>
+   <meta charset="UTF-8">
+   <title>Line Chart iframe</title>
+   <script src="chrome-untrusted://resources/js/load_time_data_deprecated.js"></script>
```
Comment (kdenhartog) [inline components/brave_wallet_ui/line_chart/line_chart.html:10]: I believe there's a new non deprecated resource we should be using here with .m.js

**Review by kdenhartog — APPROVED** [review]
Few quick questions and nits, but nothing blocking.

File: browser/ui/webui/brave_wallet/line_chart/line_chart_ui.cc [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                                              kLineChartDisplayGeneratedSize),
+                              IDR_BRAVE_WALLET_LINE_CHART_DISPLAY_HTML);
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      std::string("script-src 'self' chrome-untrusted://resources "
+                  "chrome-untrusted://brave-resources;"));
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::StyleSrc,
+      std::string(
+          "style-src 'self' 'unsafe-inline' chrome-untrusted://resources;"));
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply browser/ui/webui/brave_wallet/line_chart/line_chart_ui.cc:52]: This is for styled-components. Not functional without this.

File: browser/ui/webui/brave_wallet/line_chart/line_chart_ui.cc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  untrusted_source->UseStringsJs();
+  untrusted_source->AddString("braveWalletLineChartBridgeUrl",
+                              kUntrustedLineChartURL);
+  untrusted_source->AddString("braveWalletTrezorBridgeUrl",
+                              kUntrustedTrezorURL);
+  untrusted_source->AddString("braveWalletLedgerBridgeUrl",
+                              kUntrustedLedgerURL);
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ImgSrc,
+      std::string("img-src 'self' https: data:;"));
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply browser/ui/webui/brave_wallet/line_chart/line_chart_ui.cc]: removed https from overrides

File: components/brave_wallet_ui/line_chart/components/custom_tooltip.tsx (hunk 2 above)
Comment (josheleonard) [reply components/brave_wallet_ui/line_chart/components/custom_tooltip.tsx]: reverted back to original css code.

File: components/brave_wallet_ui/line_chart/line_chart.html (hunk 3 above)
Comment (josheleonard) [reply components/brave_wallet_ui/line_chart/line_chart.html]: Removed. It was not needed

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end

**Review by mkarolin — APPROVED** [review]
`chromium_src` and `grd` ++

**Review by yrliou — APPROVED** [review]
core++

**Pull request merged**